src/main/java/seedu/academydirectory/logic/commands/HelpCommand.java

```diff
@@ -8,14 +10,50 @@
 public class HelpCommand extends Command {
 
     public static final String COMMAND_WORD = "help";
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Shows program usage instructions for the command in query\n"
+            + "Example: " + COMMAND_WORD + "add";
+    public static final String SHOWING_HELP_MESSAGE = "Showing help.";
+    public static final String MESSAGE_HELP_SUCCESS = "Show help for command: %1$s";
+    public static final String MESSAGE_ARGUMENTS = "Syntax: %2$s";
+    private static final String[] APPROVED_COMMAND = {
+        "add", "attendance", "clear", "delete", "edit",
+        "exit", "find", "list", "retrieve"
+    };
 
-    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Shows program usage instructions.\n"
-            + "Example: " + COMMAND_WORD;
+    private final String syntax;
 
-    public static final String SHOWING_HELP_MESSAGE = "Opened help window.";
+    public HelpCommand() {
+        this.syntax = null;
+    }
+
+    /**
+     * Constructor for the Help command
+     * @param syntax of the command needed to be clarified by the users
+     */
+    public HelpCommand(String syntax) {
+        requireAllNonNull(syntax);
+        this.syntax = syntax;
+    }
+
+    private String generateSuccessMessage(String syntax) {
+        return String.format(MESSAGE_HELP_SUCCESS, syntax);
+    }
 
     @Override
     public CommandResult execute(Model model) {
         return new CommandResult(SHOWING_HELP_MESSAGE, true, false);
     }
+
+    @Override
+    public boolean equals(Object obj) {
```

Comment:
Correct me if I'm wrong. `equals()` methods of Commands are used for testing purposes right?
Do u plan to update [HelpCommandTest.java](https://github.com/AY2122S1-CS2103T-T15-3/tp/blob/master/src/test/java/seedu/academydirectory/logic/commands/HelpCommandTest.java) in this PR?

Reply:
Ah so you will leave the testing for Milestone v1.2b right?

Reply:
Yep, I will
Or if possible I will start with the testing for next week PR